app/src/main/java/com/example/development/sakaiclientandroid/fragments/assignments/AssignmentsFragment.java

```diff
@@ -0,0 +1,205 @@
+package com.example.development.sakaiclientandroid.fragments.assignments;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.widget.SwipeRefreshLayout;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.example.development.sakaiclientandroid.NavActivity;
+import com.example.development.sakaiclientandroid.R;
+import com.example.development.sakaiclientandroid.api_models.assignments.Assignment;
+import com.example.development.sakaiclientandroid.fragments.BaseFragment;
+import com.example.development.sakaiclientandroid.models.Course;
+import com.example.development.sakaiclientandroid.models.Term;
+import com.example.development.sakaiclientandroid.utils.RutgersSubjectCodes;
+import com.example.development.sakaiclientandroid.utils.custom.TreeViewItemClickListener;
+import com.example.development.sakaiclientandroid.utils.holders.AssignmentCourseViewHolder;
+import com.example.development.sakaiclientandroid.utils.holders.AssignmentTermHeaderViewHolder;
+import com.example.development.sakaiclientandroid.utils.holders.TermHeaderViewHolder;
+import com.unnamed.b.atv.model.TreeNode;
+import com.unnamed.b.atv.view.AndroidTreeView;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.example.development.sakaiclientandroid.NavActivity.ASSIGNMENTS_TAG;
+
+public class AssignmentsFragment extends BaseFragment {
+    public static final String ASSIGNMENTS_SORTED_BY_COURSES = "ASSIGNMENTS_SORTED_BY_COURSES";
+
+    private AndroidTreeView treeView;
+    private ArrayList<ArrayList<Course>> courses;
+    private ArrayList<ArrayList<Assignment>> assignments;
+    private boolean sortedByCourses;
```

Comment:
Yes, I agree it needs to be split up, especially since we might also have calendar view in the future and that would require splitting this up inevitably.